tensorrt_llm/_torch/pyexecutor/py_executor.py

```diff
@@ -1258,9 +1258,19 @@ def _fetch_new_requests(self) -> List[RequestQueueItem]:
             max(all_ranks_num_active_requests),
```

Comment:
Funtion `_fetch_new_requests` is getting long and hard to follow. We should probably split it, and have good dedicated unit tests for it, to cover both `enable_attention_dp` and ` not enable_attention_dp` cases.